exceptions/include/peach_library/exceptions/utils/exception_format.hpp

```diff
@@ -12,19 +12,29 @@
 namespace peach::detail
 {
   template< OutputStreamInsertable... Tys >
-  std::string format_error( const std::size_t line_num, const std::string& file_name,
-                            Tys&&... args )
+  std::string format_error( const int line_num, const std::string& file_name,
+                            const std::string& prefix, Tys&&... args )
   {
     std::stringstream ss { };
 
     const time_t t { std::time( nullptr ) };
     tm current_time { };
     localtime_s( &current_time, &t );
 
-    ss << std::put_time( &current_time, "[%T]" )
-       << " at (line:file) " << line_num << " : " << file_name << " ->\n\t\t{ ";
-    ( ( ss << std::forward< Tys >( args ) << ' ' ), ... );
-    ss << " }\n\n";
+    const auto put_time = std::put_time( &current_time, "[%T]" );
+
+    if ( line_num == constants::kNoSourceLocation && file_name.empty( ) )
```

Comment:
this is not the most elegant way, any alternatives?

Reply:
Isnt it enough to check for `constants::kNoSourceLocation`?

Also you could split the passing into the stringstream into multiple lines to avoid duplicate code in the `if` and `else` body.

Reply:
As for the duplicate code, theres no clean way to really do what you suggest - as both versions differ in quite some points somewhere in the middle of the stringstream. Do you have a suggestion on how to do it cleanly?

Reply:
something like this should work:
```cpp
    ss << '[' << prefix << ']' << put_time;

    if ( line_num == constants::kNoSourceLocation && file_name.empty( ) )
    {
      ss << " -> { ";
    } else
    {
      ss << '[' << prefix << ']' << put_time << " at (line:file) " << line_num << " : " << file_name
         << " ->\n\t\t{ ";
    }

    ( ( ss << std::forward< Tys >( args ) << ' ' ), ... );
    ss << " }\n\n";
```